### ISSUE TITLE
Request for shortcut property to link an Intentional Act to its Objective

### ISSUE THREAD

**Issue by mark-jensen** [CommonCoreOntology/CommonCoreOntologies#155]
There is a term [request](https://opensource.ieee.org/cyber-ontology-working-group/cyber-ontology-releases/-/issues/4) to add an object property labeled `has objective` to the Cyber Ontology, which is maintained on IEEE. The scope of this term is more general than Cyber and having a shortcut like this may prove helpful to other users of CCO. So, I am bumping that request up here to see what others think about adding it to CCO. 

Copied from linked issue on C3O tracker, submitted by Brian Haugh:
>A property to relate a 'cco:Intentional Act' to its cco:Objective is needed. More specifically, we need this type of property to relate a cco:CyberAttack to its' cco:Objective. This is needed by any application using the MITRE ATT&CK Matrix Ontology (MAMO) to relate MITRE ATT&CK Techniques to their Tactics (Tactical Objectives). Such a property (has_objective) is already in use by MAMO and the Ontology for Attacks in Cyber Risk Assessment (OACRA). We recommend adding this property to the Cyber Ontology, where it is more generally accessible to other ontologies and applications. As an alternative, this property might be added to the CCO Event ontology, although that might take longer since its IEEE standardization effort has not yet begun. 

cco:has_objective
domain cco:IntentionalAct 
range cco:Objective
rdfs:label "has objective"
cco:definition "x has objective y iff x is an 'Intentional Act', and y is an 'Objective', and z is a 'Plan', and z 'has part' y, and z 'prescribes' x."
cco:elucidation "An Intentional Act has an Objective if that act is prescribed by a Plan which has part the Objective. The agent carrying out the act does so according to the plan so as to achieve the Objective."
owl property chain: 'prescribed by' o 'has part' <-- 'has objective'

**Comment by neilotte** [comment]
I think it would be helpful if we started a short cut relations ontology to accompany our core set of relations, where each short cut relation was accompanied by a transformation query that unpacks the shortcut into the larger recommendation. Thoughts?

**Comment by swartik** [comment]
In a conversation last week someone mentioned the possibility of achieving an objective by doing nothing. Does abstaining from acting count as an Intentional Act?

**Comment by cameronmore** [comment]
@neilotte would we need a whole ontology or just a brief document that describes them? Depending on how many shortcuts you have, is it worth it? Open to the idea, but there should be a part in each module's documentation that explains the shortcuts, assuming a module or ontology is documented/well documented.

**Comment by neilotte** [comment]
@cameronmore Let me withdraw my suggestion and instead offer that we just need a way of curating shortcuts so that it is clear that they are published AS shortcuts (separate from foundational relations) and so that the link to their transformations is clear. Whether that happens in one or more files is incidental. Does that make sense? I'm also open to suggestions here.

**Comment by cameronmore** [comment]
@neilotte Agree, we discussed this in the IEEE meeting you missed, and I personally favor the policy "shortcuts are fine as long as the robust semantics do exist outside PowerPoints in an owl file somewhere".  We did not rule on specifics with respect to short cuts and which ones are permitted v which ones are not because it speaks to a larger issue of 'did you do good modeling or not.' Hope we will discuss the shortcut question in the next few weeks in our meetings.

**Comment by dlutz2** [comment]
Assuming that "robust semantics" means a SPARQL query or FOL fragment, is it sufficient to have an annotation property which links the property (or axiom?) with that query or FOL fragment? Could be in the same declaring OWL file, an overlay, Big List o' Shortcuts Ontology ...

**Comment by cameronmore** [comment]
I'm thinking along the lines of something like the following paradigmatic case in the information model:

Rather than...

`cco:Person cco:designated_by cco:CodeIdentifier .
cco:CodeIdentifier obo:generically_depends_on cco:InformationBearingEntity .
cco:InformationBearingEntity cco:has_value "A123" .`

We can simply assert...

`cco:Person cco:designated_by cco:CodeIdentifier .
cco:CodeIdentifier cco:is_tokenized_by "A123" .`

Skipping the IBE. A hypothetical might be "Cameron drives car" means "Cameron agent in Act of Vehicle Use".
